In [1]:
import pandas as pd
import numpy as np

### Import CSV files

In [2]:
# TRAIN
train_month_1 = pd.read_csv('train_month_1.csv')
train_month_2 = pd.read_csv('train_month_2.csv')
train_month_3_with_target = pd.read_csv('train_month_3_with_target.csv')

# TEST
test_month_1 = pd.read_csv('test_month_1.csv')
test_month_2 = pd.read_csv('test_month_2.csv')
test_month_3 = pd.read_csv('test_month_3.csv')

## Data Discovery

In [9]:
train_month_1.shape

(63697, 39)

In [8]:
# Find NaN value counts per column
train_month_1.isna().sum()

client_id                                0
homebanking_active                       0
has_homebanking                          0
has_insurance_21                         0
has_insurance_23                         0
has_life_insurance_fixed_cap             0
has_life_insurance_decreasing_cap        0
has_fire_car_other_insurance             0
has_personal_loan                        0
has_mortgage_loan                        0
has_current_account                      0
has_pension_saving                       0
has_savings_account                      0
has_savings_account_starter              0
has_current_account_starter              0
bal_insurance_21                         0
bal_insurance_23                         0
cap_life_insurance_fixed_cap             0
cap_life_insurance_decreasing_cap        0
prem_fire_car_other_insurance            0
bal_personal_loan                        0
bal_mortgage_loan                        0
bal_current_account                      0
bal_pension

## Insights

### train_month_1

* shape: (63697, 39)
* `customer_education` can be dropped since it has `47125` **NaN** value

## TODO

- [ ] Convert string values to numbers: `customer_children`, `customer_relationship`...
- [ ] Learn a way to use dates, maybe convert them to years by ignoring the month?
- [ ] Normalize `customer_gender`
- [ ] Decide what to do with `customer_postal_code`

In [36]:
train_month_3_with_target

client_id  homebanking_active  has_homebanking  \
0      910df42ad36243aa4ce16324cd7b15b0                   0                0   
1      4e19dc3a54323c5bbfc374664b950cd1                   1                1   
2      f5d08db1b86c0cb0f566bf446cff1fb4                   1                1   
3      26170ecf63653e215c52f4262c1c4859                   0                0   
4      c078009957dffb64f20e61b41220a976                   0                0   
...                                 ...                 ...              ...   
63692  0a58f2eb841ddac0626dacac6ca69524                   1                1   
63693  193be2222be99bf04f42193b5cdfb95d                   0                0   
63694  fa9f074ec8cad610ccaec2270021490e                   0                1   
63695  5236064e97c655b0ad99ed2155e1139e                   0                0   
63696  977dda870c3f54df46297df3869b2907                   1                1   

       has_insurance_21  has_insurance_23  has_life_insurance_fixed_cap  \
0                     0                 0                             0   
1                     0                 0                             0   
2                     0                 0                             0   
3                     0                 0                             0   
4                     0                 0                             0   
...                 ...               ...                           ...   
63692                 1                 0                             0   
63693                 1                 0                             0   
63694                 0                 0                             0   
63695                 0                 0                             0   
63696                 0                 0                             0   

       has_life_insurance_decreasing_cap  has_fire_car_other_insurance  \
0                                      0                             1   
1                                      0                             0   
2                                      0                             1   
3                                      0                             1   
4                                      0                             0   
...                                  ...                           ...   
63692                                  0                             0   
63693                                  0                             1   
63694                                  1                             1   
63695                                  0                             0   
63696                                  0                             0   

       has_personal_loan  has_mortgage_loan  ...  customer_since_bank  \
0                      0                  0  ...              1994-08   
1                      0                  0  ...              2017-01   
2                      0                  0  ...              1980-12   
3                      0                  0  ...              2013-10   
4                      0                  0  ...              2012-11   
...                  ...                ...  ...                  ...   
63692                  0                  0  ...              1998-06   
63693                  0                  1  ...              1989-05   
63694                  0                  1  ...              1991-01   
63695                  0                  0  ...              2014-04   
63696                  0                  0  ...              2002-02   

       customer_gender  customer_birth_date  customer_postal_code  \
0                    1              1943-09                  3630   
1                    1              1994-02                  2460   
2                    2              1936-10                  2660   
3                    1              1946-09                  6600   
4                    2              1996-04              

In [38]:
train_month_3_with_target.describe()

homebanking_active  has_homebanking  has_insurance_21  \
count        63697.000000     63697.000000      63697.000000   
mean             0.215191         0.280939          0.095028   
std              0.410958         0.449462          0.293256   
min              0.000000         0.000000          0.000000   
25%              0.000000         0.000000          0.000000   
50%              0.000000         0.000000          0.000000   
75%              0.000000         1.000000          0.000000   
max              1.000000         1.000000          1.000000   

       has_insurance_23  has_life_insurance_fixed_cap  \
count      63697.000000                  63697.000000   
mean           0.009953                      0.002747   
std            0.099270                      0.052344   
min            0.000000                      0.000000   
25%            0.000000                      0.000000   
50%            0.000000                      0.000000   
75%            0.000000                      0.000000   
max            1.000000                      1.000000   

       has_life_insurance_decreasing_cap  has_fire_car_other_insurance  \
count                       63697.000000                  63697.000000   
mean                            0.111779                      0.318053   
std                             0.315097                      0.465724   
min                             0.000000                      0.000000   
25%                             0.000000                      0.000000   
50%                             0.000000                      0.000000   
75%                             0.000000                      1.000000   
max                             1.000000                      1.000000   

       has_personal_loan  has_mortgage_loan  has_current_account  ...  \
count       63697.000000       63697.000000         63697.000000  ...   
mean            0.041619           0.098089             0.500809  ...   
std             0.199718           0.297438             0.500003  ...   
min             0.000000           0.000000             0.000000  ...   
25%             0.000000           0.000000             0.000000  ...   
50%             0.000000           0.000000             1.000000  ...   
75%             0.000000           0.000000             1.000000  ...   
max             1.000000           1.000000             1.000000  ...   

       bal_savings_account_starter  bal_current_account_starter  \
count                 63697.000000                 63697.000000   
mean                     57.641176                    30.320894   
std                     892.959859                   407.877892   
min                       0.000000                  -330.000000   
25%                       0.000000                     0.000000   
50%                       0.000000                     0.000000   
75%                       0.000000                     0.000000   
max                   24050.000000                 19790.000000   

       visits_distinct_so  visits_distinct_so_areas  customer_gender  \
count        63697.000000              63697.000000     63697.000000   
mean             1.230199                  1.042608         1.486444   
std              0.501498                  0.224991         0.499820   
min              1.000000                  1.000000         1.000000   
25%              1.000000                  1.000000         1.000000   
50%              1.000000                  1.000000         1.000000   
75%              1.000000                  1.000000         2.000000   
max              7.000000                  6.000000         2.000000   

       customer_postal_code  customer_occupation_code  customer_self_employed  \
count          63697.000000              61695.000000            63697.000000   
mean            5577.261959                  8.773531                0.087021   
std             3020.064554                  1.131453                0.281869   
min           

In [21]:
# Most of the data is labeled as 0 :( 
train_month_3_with_target['target'].value_counts()

0    61784
1     1913
Name: target, dtype: int64